src/Entity/Idea.php

```diff
@@ -164,4 +169,26 @@ public function removeSupporter(User $supporter): static
 
         return $this;
     }
+
+    public function getEndDate(): ?\DateTimeImmutable
+    {
+        return $this->endDate;
+    }
+
+    public function setEndDate(\DateTimeImmutable $endDate): static
+    {
+        $this->endDate = $endDate;
+
+        return $this;
+    }
+
+    public function getRemainingDays(Idea $idea): int
+    {
+        $endDate = $idea->getEndDate();
```

Comment:
do you really need an idea here ? IN an idea ?? Couldn't $this wouldn't be enough ?